src/common/snippets/src/pass/extract_reshapes_from_mha.cpp

```diff
@@ -66,3 +66,87 @@ ov::snippets::pass::ExtractReshapesFromMHA::ExtractReshapesFromMHA() {
     auto m = std::make_shared<pattern::Matcher>(reshape_2_m, matcher_name);
     register_matcher(m, callback);
 }
+
+ov::snippets::pass::RankUpgradeToRankReduction::RankUpgradeToRankReduction() {
+    MATCHER_SCOPE(RankUpgradeToRankReduction);
+    auto static_shape_single_consumer = [](const ov::Output<ov::Node>& out) {
+        return pattern::has_static_shape()(out) && pattern::consumers_count(1)(out);
+    };
+    // reshape_1_m insert leading dimension of 1.
+    auto rank_upgrade_reshape = [&](const ov::Output<ov::Node>& out) {
+        if (!static_shape_single_consumer(out)) {
+            return false;
+        }
+        auto out_shape = out.get_shape();
+        if (out_shape.size() < 1 || out_shape[0] != 1) {
+            return false;
+        }
+        out_shape.erase(out_shape.begin());
+        const auto& in_shape_partial = out.get_node_shared_ptr()->get_input_partial_shape(0);
+        if (in_shape_partial.is_dynamic()) {
+            return false;
+        }
+        const auto& in_shape = in_shape_partial.to_shape();
+        return out_shape == in_shape;
+    };
+    // input_2_m has leading dimension of 1.
+    auto has_leading_dimension_one = [&](const ov::Output<ov::Node>& out) {
+        if (!static_shape_single_consumer(out)) {
+            return false;
+        }
+        const auto& out_shape = out.get_shape();
+        return ((out_shape.size() > 0) && (out_shape[0] == 1));
+    };
+    // reshape_2_m delete leading dimension of 1.
+    auto rank_reduction_reshape = [&](const ov::Output<ov::Node>& out) {
+        if (!static_shape_single_consumer(out)) {
+            return false;
+        }
+        auto in_shape_partial = out.get_node_shared_ptr()->get_input_partial_shape(0);
+        if (in_shape_partial.is_dynamic()) {
+            return false;
+        }
+        auto in_shape = in_shape_partial.to_shape();
+        if (in_shape.size() < 1 || in_shape[0] != 1) {
+            return false;
+        }
+        in_shape.erase(in_shape.begin());
+        const auto& out_shape = out.get_shape();
+        return out_shape == in_shape;
+    };
+
+    auto matmul_m = pattern::wrap_type<opset1::MatMul>(static_shape_single_consumer);
+    auto input_1_m = pattern::any_input(pattern::has_static_shape());
+    auto add_1_m = pattern::wrap_type<opset1::Add>({matmul_m, input_1_m}, static_shape_single_consumer);
```

Comment:
1. `add_1_m` seems like not necessary: if it is absent, the transformation is still valid. Can we make it optional (`pattern::optional`)?
2. It seems like the transformation is valid not only in case of Add, it can be generalized on all binary eltwises. Can we use   base class for binary eltwises in the matcher instead of Add?

Reply:
Yes, we can extend to cover more cases for both type and presence, have done. thanks Vladislav!